src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -384,6 +483,7 @@ private void recordLineage(BatchSinkContext context,
   }
 
   private LegacySQLTypeName getTableDataType(Schema schema) {
+    schema = BigQueryUtil.getNonNullableSchema(schema);
```

Comment:
does this mean there was a bug with nullable schemas before this change?

Reply:
No. Before we were calling `BigQueryUtil.getNonNullableSchema` before  `getTableDataType ` method was called. However now with array types, we are using component schema so calling `getTableDataType ` method recursively. Thats why we need to check here now.